src/vehicles/RosVehicle.lua

```diff
@@ -0,0 +1,257 @@
+--[[
+Copyright (c) 2021, TU Delft
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+author: Ting-Chia Chiang
+author: G.A. vd. Hoorn
+--]]
+-- source(Utils.getFilename("src/modROS.lua", g_modROSModDirectory))
+
+RosVehicle = {}
+
+RosVehicle.MOD_DIR = g_modROSModDirectory
+RosVehicle.MOD_NAME = g_modROSModName
+
+
+function RosVehicle.prerequisitesPresent()
+    return true
```

Comment:
but for those attachable tools like plows or seeders are not drivable, and (if) we want to get the odom data of them.

Reply:
Hm.

For now I would say lets stick to drivable types. Do we have a check in-place which makes sure the ROS-side is not trying to control an implement?

We can always lower requirements at a later time if that would make sense.

Reply:
There is only a simple[ condition/check](https://github.com/tckarenchiang/FS19_modROS/blob/add_spec_RosVehicle/src/modROS.lua#L281) to prevent ROS from controlling an implement. I just make sure the vehicle is drivable before sending the input values.